File: examples/python/create_payment.py
import json
import os
import urllib.error
import urllib.request

base_url = os.getenv("XPAYR_BASE_URL", "https://xpayr.com/api/v1").rstrip("/")
secret_key = os.getenv("XPAYR_SECRET_KEY")
if not secret_key:
    raise RuntimeError("Set XPAYR_SECRET_KEY to an sk_test_* key")

payload = json.dumps({
    "amount": "49.90",
    "currency": "USDC",
    "network": "bsc-testnet",
    "order_id": "ORDER-1001",
    "description": "Test checkout",
    "metadata": {"source": "xpayr-api-examples"},
}).encode("utf-8")

request = urllib.request.Request(
    f"{base_url}/payments",
    data=payload,
    headers={"Authorization": f"Bearer {secret_key}", "Content-Type": "application/json"},
    method="POST",
)

try:
    with urllib.request.urlopen(request, timeout=20) as response:
        print(response.read().decode("utf-8"))
except urllib.error.HTTPError as error:
    raise RuntimeError(f"XPayr API {error.code}: {error.read().decode('utf-8')}") from error

File: examples/node/create-payment.mjs
import { buildPaymentRequest } from "../../src/payment-request.mjs";

const baseUrl = process.env.XPAYR_BASE_URL || "https://xpayr.com/api/v1";
const secretKey = process.env.XPAYR_SECRET_KEY;
if (!secretKey) throw new Error("Set XPAYR_SECRET_KEY to an sk_test_* key");

const response = await fetch(`${baseUrl}/payments`, {
  method: "POST",
  headers: {
    Authorization: `Bearer ${secretKey}`,
    "Content-Type": "application/json",
  },
  body: JSON.stringify(buildPaymentRequest({
    amount: "49.90",
    currency: "USDC",
    network: "bsc-testnet",
    orderId: "ORDER-1001",
    description: "Test checkout",
    metadata: { source: "xpayr-api-examples" },
  })),
});

const payload = await response.json();
if (!response.ok) throw new Error(`XPayr API ${response.status}: ${JSON.stringify(payload)}`);
console.log(JSON.stringify({ id: payload.id, payment_url: payload.payment_url, status: payload.status }, null, 2));
